File: Array Algorithms/FindAlgorithms/FindTargetInArray.java
package FindAlgorithms;

public class FindTargetInArray {

    /*
        by @heiphin7
        Алгоритм поиска значения в массиве, которая работает за O(n) - линейное время работы
    */

    public static int findInArray(int[] array, int target) {
        // Сравниваем каждый элемент массива с нужным
        for (int i = 0; i < array.length; i++) {
            if (array[i] == target) {
                // Если текущий элемент в массиве равен цели - возвращаем его
                return i;
            }
        }

        // Если значение не найдено, возвращаем -1
        return -1;
    }
}

File: Array Algorithms/FindAlgorithms/cpp/FindTargetInArray.cpp
#include <iostream>
#include <vector>

using namespace std;

/*
    by @heiphin7

    Нахождение индекса нужного элемента в массиве за O(n), то есть линейное время
    Цикл, который заполняет массив, не считается, так как это ввод данных
*/

int main() {
	int n, target;
	cin >> n >> target;

	vector<int> arr(n);

	for (int i = 0; i < n; ++i) {
		cin >> arr[i];
	}

	for (int i = 0; i < n; ++i) {
		if (arr[i] == target) {
			cout << "index of target in array: " << i << endl;
			return 0;
		}
	}

	cout << "target not founded";
	return 0;
}